### ISSUE TITLE
[P2] Cleanup and update

### ISSUE THREAD

**Issue by tancheng** [tancheng/VectorCGRA#39]
Deprecate en/rdy and use val/rdy for all modules.

**Comment by tancheng** [comment]
@RuihongY FYI.

**Comment by tancheng** [comment]
Hi @yo96, assigning this cleanup bug to you.

**Comment by tancheng** [comment]
The concrete TODOs:
 - [ ] Replace en/rdy with val/rdy
 - [ ] Resolve latches (provide initialization for all signals within a combinational block)
 - [ ] Update CGRA and PyOCN to latest PyMTL4.0 (or 3.0 whatever)
 - [ ] Make the xbar more modular rather than a behavioral-like design

**Comment by tancheng** [comment]
Hi @yo96, would following introduce combinational loop?
```
s.send_out[0].msg.predicate @= s.send_out[0].msg.predicate
```
in https://github.com/tancheng/VectorCGRA/blob/49a40187f90a8d67b0a410c0d3bc5944422e22e6/fu/single/AdderRTL.py#L104C9-L104C67

**Comment by yo96** [comment]
Yes, that looks like a combinational loop to me. It is like connecting the output of an AND gate to its input.

**Comment by tancheng** [comment]
> The concrete TODOs:
> 
> * [ ]  Replace en/rdy with val/rdy
> * [ ]  Resolve latches (provide initialization for all signals within a combinational block)
> * [ ]  Update CGRA and PyOCN to latest PyMTL4.0 (or 3.0 whatever)
> * [ ]  Make the xbar more modular rather than a behavioral-like design

Updated:

 - [x]  Replace en/rdy with val/rdy: https://github.com/tancheng/VectorCGRA/pull/54
 - [ ]  Resolve latches (provide initialization for all signals within a combinational block)
 - [ ]  Update CGRA and PyOCN to latest PyMTL4.0 (or 3.0 whatever)
 - [ ]  Make the xbar more modular rather than a behavioral-like design
